frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -49,6 +49,8 @@ const (
 
 	sbomScanContext = "BUILDKIT_SBOM_SCAN_CONTEXT"
 	sbomScanStage   = "BUILDKIT_SBOM_SCAN_STAGE"
+
+	keyDockerfileLintArg = "build-arg:BUILDKIT_DOCKERFILE_LINT"
```

Comment:
`_CHECK` if that is what the directive is called

Reply:
Ideally parsing this should be moved to https://github.com/moby/buildkit/blob/master/frontend/dockerui/config.go#L49-L60

Reply:
My previous comment may be confusing. The "parsing" don't necessarily need to be in dockerui, but the reading of the build arg and the `keyDockerfileLintArg` definition.

The new field in `dockerui.Config` can be either.

```
type Config struct {
	BuildArgs        map[string]string
	CacheIDNamespace string
        LinterConfig string
```
 or 
```
type Config struct {
	BuildArgs        map[string]string
	CacheIDNamespace string
        LinterConfig *linter.Config
```

whatever works best

Reply:
went with the first!